drivers/l3gxxxx/l3gxxxx.c

```diff
@@ -0,0 +1,1163 @@
+                                       break;
+#endif
+#if IS_USED(MODULE_L3GD20)
+        case L3GXXXX_CHIP_ID_L3GD20:   dev->sensor = L3GD20;
+                                       break;
+#endif
+#if IS_USED(MODULE_L3G4200D) || IS_USED(MODULE_I3G4250D) || IS_USED(MODULE_A3G4250D)
+        case L3GXXXX_CHIP_ID_X3G42XXD: dev->sensor = X3G42XXD;
+                                       break;
+#endif
+        default: DEBUG_DEV("sensor is not available, wrong chip id %02x",
+                           dev, chip_id);
+                 return -L3GXXXX_ERROR_WRONG_CHIP_ID;
+    }
+
+    return res;
+}
+
+static int _update(const l3gxxxx_t *dev,
+                   uint8_t reg, uint8_t mask, uint8_t val)
+{
+    DEBUG_DEV("reg=%02x mask=%02x val=%02x", dev, reg, mask, val);
+
+    int res = L3GXXXX_OK;
+
+    uint8_t reg_val;
+    uint8_t shift = 0;
+
+    while (!((mask >> shift) & 0x01)) {
+        shift++;
+    }
+
+    /* read current register value */
+    res |= _read(dev, reg, &reg_val, 1);
+
+    /* set masked bits to the given value  */
+    reg_val = (reg_val & ~mask) | ((val << shift) & mask);
+
+    /* write back new register value */
+    res |= _write(dev, reg, &reg_val, 1);
+
+    return res;
+}
+
+#define L3GXXXX_SPI_BUF_SIZE       64   /* SPI register data buffer size */
+#define L3GXXXX_SPI_READ_FLAG      0x80
+#define L3GXXXX_SPI_WRITE_FLAG     0x00
+#define L3GXXXX_SPI_AUTO_INC_FLAG  0x40
+
+#define L3GXXXX_I2C_AUTO_INC_FLAG  0x80
+
+static void _acquire(const l3gxxxx_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+#if IS_USED(MODULE_L3GXXXX_SPI)
+    if (dev->params.if_params.type == L3GXXXX_SPI) {
+        spi_acquire(_SPI_DEV, _SPI_CS, SPI_MODE_3, _SPI_CLK);
+        return;
+    }
+#endif
+#if IS_USED(MODULE_L3GXXXX_I2C)
+    if (dev->params.if_params.type == L3GXXXX_I2C) {
+        i2c_acquire(_I2C_DEV);
+        return;
+    }
+#endif
+    return;
+}
+
+static void _release(const l3gxxxx_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+#if IS_USED(MODULE_L3GXXXX_SPI)
+    if (dev->params.if_params.type == L3GXXXX_SPI) {
+        spi_release(_SPI_DEV);
+        return;
+    }
+#endif
+#if IS_USED(MODULE_L3GXXXX_I2C)
+    if (dev->params.if_params.type == L3GXXXX_I2C) {
+        i2c_release(_I2C_DEV);
+        return;
+    }
+#endif
+    return;
+}
+
+static int _read(const l3gxxxx_t *dev, uint8_t reg, uint8_t *data, uint8_t len)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(data != NULL);
+    ASSERT_PARAM(len != 0);
+
+    DEBUG_DEV("read %d byte from sensor registers starting at addr 0x%02x",
+              dev, len, reg);
+
+#if IS_USED(MODULE_L3GXXXX_SPI)
+    if (dev->params.if_params.type == L3GXXXX_SPI) {
+        ASSERT_PARAM(len < L3GXXXX_SPI_BUF_SIZE)
+
+        reg &= 0x3f;
+        reg |= L3GXXXX_SPI_READ_FLAG;
+        reg |= L3GXXXX_SPI_AUTO_INC_FLAG;
+
+        /* prepare input and output by stream */
+        static uint8_t mosi[L3GXXXX_SPI_BUF_SIZE];
+        static uint8_t miso[L3GXXXX_SPI_BUF_SIZE];
```

Comment:
Why are those static buffers needed here?

Reply:
Removed by splitting the transfer into
```
spi_transfer_byte(_SPI_DEV, _SPI_CS, true, reg);
spi_transfer_bytes(_SPI_DEV, _SPI_CS, false, 0, data, len);
```